cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -284,13 +294,11 @@ class LeadImpl extends ServerImpl with Lead
       SnappyTableStatsProviderService.start(sc, url = null)
 
       if (startHiveServer) {
-        val useHiveSession = Property.HiveServerUseHiveSession.get(conf)
         val hiveService = SnappyHiveThriftServer2.start(useHiveSession)
-        val sessionKind = if (useHiveSession) "session=hive" else "session=snappy"
-        SnappyHiveThriftServer2.getHostPort(hiveService) match {
-          case None => addStartupMessage(s"Started hive thrift server ($sessionKind)")
+        if (jobServerWait) SnappyHiveThriftServer2.getHostPort(hiveService) match {
```

Comment:
How does it start in background?

Reply:
Earlier code had: jobserverWait = startHiveServer || Property.JobServerWaitForInit.get(conf). So when hive server was configured to start, then it used to wait for entire initialization to finish. Now the new variable "startHiveServerDefault" will be true if nothing has been configured and in that case jobServerWait will be false.

So the code added before that adds startup message when startHiveServerDefault is true will display the message but the host/port won't be shown in that case because it is obtained only after startup of hive server (which can read from SparkConf or HadoopConf).

Reply:
Looked like start() call above is blocking. Just that some service may have been invoked in the background so no information of host:port immediately.